### PR TITLE
Standardize cutscene and victory UX

### LINKED ISSUE #5514: Standardize victory cues
### Game and System Information
Battle for Wesnoth v1.15.9+dev x86_64

### Describe the desired feature
The vast majority of the scenarios I have played so far have trained me to expect certain visual and auditory cues when I win a scenario. This includes:

* The music changes to victory.ogg or victory2.ogg
* There is a popup telling me that I won

In the absence of the cues it takes me a bit to realize that the scenario did in fact end. This inconsistency is surprising and confusing.

#### Examples (this is not an exhaustive list)
Tutorial: S1: no victory popup, no music change
AToTB: 04_Return_to_the_Village: no victory popup, no music change
TSG: 08a_Vengeance: no victory popup, no music change
TSG: 08b_The_Tides_of_War: no victory popup, no music change
Liberty: 07_Glory: no victory popup, no music change
HttT: 24_Battle_for_Wesnoth: no victory popup, no music change

### PR DESCRIPTION
**Pull request by hexagonrecursion**
Fixes #5514

Your user experience gives me as a player certain expectations:
* When a cutscene ends the game automatically moves on to the next scenario without a victory popup or linger mode
* When a regular scenario ends:
  * The music changes to victory.ogg or victory2.ogg
  * A victory popup is shown
  * The game enters the linger mode

Scenarios that violate this expectations surprise and confuse me. This PR fixes every such scenario that I was able to find in the mainline campaigns.

### CODE REVIEW

**Review by knyghtmare — COMMENTED** [review]
My review is mostly constructive and I have added reasoning for any changes.

- Dead Water S13 (Epilogue): This is fine as I recall this is the last scenario.
- DM (S22): I don't think a carryover report should be shown in this scenario as it is clearly the last scenario of the campaign
- DiD (S7abc): These scenarios are meant to be a single continuation of Malin and Volk's mission to "retrieve" the book. I don't think carryover reports should be displayed at all here as there is no recruiting after Scenario 7A, and the gold is fully carried over.
- S10 (Endless Night): I see changes have been made to the "secret" endings. I don't see any point in displaying the carryover report here as this is the end of the campaign (and S10's loop) since it's not relevant to the player what gold would be carried over since `next_scenario=null`. I suppose linger can be shown.
- EI (final_battle utils): Same argument given in my previous case for S10 of DiD. I do agree with it having linger mode though.
- HttT (S24): Since the campaign ends with the Battle for Wesnoth, is there any point in showing how much gold the player gets carried over to an epilogue? I do not think so.
- LoW (S23): Same argument given for my previous 3 cases.
- Liberty (S7): Same argument made in my previous 4 cases. 
- NR (S13a), SoF S07, SoF S09, SotA S21, SotBE S18, THoT S09, TRoW S22, TSG S08a, TSG S08b, AToTB S04, UtBS S12: I listed all of these together as they all seem to be the same aforementioned argument.
- UtBS S07a and S07b: This was rather just a "recall someone to pick the sword" scenario and a cutscene so disabling linger mode here is perfectly fine.
- Battle Training (tutorial) S1: The `carryover_percentage=0` sort of nullifies any benefit from seeing a carryover report. Although a counter argument can be offered to allow new players to experience what a carryover report and point out some things in it, like how the values are calculated and when to consider getting the early finish bonus (over more XP for units) and retaining gold.

I have offered my feedback on these changes based on being a part of this community since 2010 and as an UMC author.

**Comment by Vultraz** [comment]
I very much disagree that there always need be a carryover report shown. Some scenarios - especially the DiD ones, for instance - are explicitly designed to, as knyghtmare said, flow into one another smoothly and feel like one scenario. The use (or lack thereof) of carryover reports is almost always a creative decision and not an oversight, and so I would oppose merging this pr as-in.

**Comment by hexagonrecursion** [comment]
The developer's / content creator's perspective: we wanted to make the gold carryover mechanic more obvious to the player. We added a popup to notify the player about it. There is no carryover in the last scenario of a campaign therefore there is no point in showing the popup.

Th player's perspective: when I win a scenario the music changes I see a popup telling me that I won. I expect this to happen every time I win a scenario.

**Comment by Pentarctagon** [comment]
Repeating your opinion again will not by itself make it more convincing.

**Comment by hexagonrecursion** [comment]
I did add something new this time. I have said: you are thinking like developers. Put yourself in the player's shoes.

**Comment by Vultraz** [comment]
Except we are. We're considering the player experience, not favoring consistency for consistency's sake.

**Comment by CelticMinstrel** [comment]
> The developer's / content creator's perspective: we wanted to make the gold carryover mechanic more obvious to the player. We added a popup to notify the player about it. There is no carryover in the last scenario of a campaign therefore there is no point in showing the popup.

But I think it does makes sense not to show information that is irrelevant. If there's no gold carryover, then why would you show the gold carryover information?

> Th player's perspective: when I win a scenario the music changes I see a popup telling me that I won. I expect this to happen every time I win a scenario.

On the other hand, this also makes a lot of sense. The popup _is_ a pretty good indication that you won.

But maybe there is a compromise solution. Logically, that WML key specifies whether to show the carryover report, _not_ whether to show the popup. Why not just show the popup _without_ including the gold carryover report in it? I don't know if there's anything else shown in the popup besides gold carryover, but you could easily add a generic "Victory!" message or something.

File: data/campaigns/Descent_Into_Darkness/scenarios/07a_A_Small_Favor.cfg [hunk 1]
```diff
@@ -770,7 +770,6 @@
         [/kill]
         [endlevel]
             result=victory
-            linger_mode=no
```
Comment (nemaara) [inline data/campaigns/Descent_Into_Darkness/scenarios/07a_A_Small_Favor.cfg:773]: Leave the DiD ones as they were originally please, including the ones for Endless Night.

File: data/campaigns/Under_the_Burning_Suns/scenarios/12_The_Final_Confrontation.cfg [hunk 2]
```diff
@@ -1567,8 +1567,6 @@
         [endlevel]
             result=victory
             bonus=no
-            carryover_report=no
```
Comment (nemaara) [inline data/campaigns/Under_the_Burning_Suns/scenarios/12_The_Final_Confrontation.cfg:1570]: Leave this one as the original as well please.

**Review by nemaara — CHANGES_REQUESTED** [review]

**Comment by hexagonrecursion** [comment]
@Wedge009

What caused me to open this issue is a specific subjective experience: when I won TSG and what I expected to happen did not happen I was confused. I asked myself: "Is this it or is there going to be more? Did the game break?" I attribute this confusion to me being a new player.

> But maybe there is a compromise solution. Logically, that WML key specifies whether to show the carryover report, _not_ whether to show the popup. Why not just show the popup _without_ including the gold carryover report in it?

This is indeed a better solution. I have somehow missed the fact that carryover information is currently always included in the popup. This sounds like something that would have to be implemented in C++ rather than wml

**Comment by Wedge009** [comment]
Thinking about the discussion from #5514, I remember that some decisions regarding scenario transitions are for story reasons, as others have already mentioned here. And unlike some, I see things more as a player than a developer so I understand what you're saying, but I don't necessarily agree with need for a pop-up. The grey tint of 'linger mode' and/or the victory music would be sufficient cues for me, but if you really feel the need for a pop-up of some sort, I'd agree with celmin's suggestion.

**Comment by Pentarctagon** [comment]
At a minimum though, are the requested changes going to be made to this PR?

**Comment by Pentarctagon** [comment]
Closing due to lack of response to nemaara's feedback.

**Pull request closed without merging**